lib/matplotlib/artist.py

```diff
@@ -54,6 +56,9 @@ def draw_wrapper(artist, renderer):
                 renderer.stop_filter(artist.get_agg_filter())
             if artist.get_rasterized():
                 renderer._raster_depth -= 1
+                if renderer._raster_depth == 0 and renderer._rasterizing:
+                    renderer.stop_rasterizing()
+                    renderer._rasterizing = False
```

Comment:
This effectively prevents merging rasterized artists together in all cases which I do not think we want to do.

Reply:
Yes.  If we want to keep this behavior and still want to fix #23999, I guess we need to make the draw method of ALL artists to be decorated somehow.